consensus/hotstuff/follower_loop.go

```diff
@@ -45,27 +50,36 @@ func NewFollowerLoop(log zerolog.Logger, followerLogic FollowerLogic) (*Follower
 	return fl, nil
 }
 
-// SubmitProposal feeds a new block proposal (header) into the FollowerLoop.
-// This method blocks until the proposal is accepted to the event queue.
+// AddCertifiedBlock appends the given certified block to the tree of pending
+// blocks and updates the latest finalized block (if finalization progressed).
+// Unless the parent is below the pruning threshold (latest finalized view), we
+// require that the parent has previously been added.
 //
-// Block proposals must be submitted in order, i.e. a proposal's parent must
-// have been previously processed by the FollowerLoop.
-func (fl *FollowerLoop) SubmitProposal(proposal *model.Proposal) {
+// Notes:
+//   - Under normal operations, this method is non-blocking. The follower internally
+//     queues incoming blocks and processes them in its own worker routine. However,
+//     when the inbound queue is, we block until there is space in the queue. This
+//     behaviours is intentional, because we cannot drop blocks (otherwise, we would
```

Comment:
```suggestion
//     behavior is intentional, because we cannot drop blocks (otherwise, we would
```

Reply:
done in https://github.com/onflow/flow-go/pull/4234/commits/859cbc8f8b5a9a95447ee0d11c949d2e2de9612b